src/Pate/Memory/MemTrace.hs

```diff
@@ -654,7 +684,9 @@ execMacawStmtExtension (MacawArchEvalFn archStmtFn) mkundef syscallModel mvar gl
     MacawArchStmtExtension archStmt -> archStmtFn mvar globs archStmt
 
     MacawArchStateUpdate{} -> \cst -> pure ((), cst)
-    MacawInstructionStart{} -> \cst -> pure ((), cst)
+    MacawInstructionStart _baddr iaddr dis ->
+      liftToCrucibleState mvar $ \sym ->
+        modify (\mem -> mem{ memCurrentInstr = toMuxTree sym (Just (iaddr,dis)) })
```

Comment:
@travitch, Not sure I'm grabbing the right data here, sometimes the results I get are a bit strange.  What's the difference between the "block address" and the "instruction address" here?

Reply:
Ah, that makes a lot more sense.